versions-common/src/main/java/org/codehaus/mojo/versions/api/ArtifactVersions.java

```diff
@@ -218,6 +217,12 @@ public ArtifactVersion[] getVersions(boolean includeSnapshots) {
                         .toArray(ArtifactVersion[]::new);
     }
 
+    public boolean isEmpty(boolean includeSnapshots) {
+        return includeSnapshots
+                ? versions.isEmpty()
+                : versions.stream().map(Object::toString).allMatch(ArtifactUtils::isSnapshot);
```

Comment:
if includeSnapshots is true we check if any versions are present - nevermind snapshot or not 
but if includeSnapshots is false we check if all versions are snapshots ...

maybe some comments for this methods - what does do

Reply:
This means -- if include snapshots is false -- then **if there are any versions that are there, all of them are snapshots**

Reply:
I'll add the following comment:

```
  // the below means: the only versions that could be present in the collection
  // are snapshots
```

Reply:
Also added javadoc. 

Instead of this, `getVersions().length == 0` was used, which acted the same.